### PR TITLE
Implement `unpack2x16float` tests

### LINKED ISSUE #1293: wgsl: execution test for unpack2x16float builtin
```
@const fn unpack2x16float(e: u32) -> vec2<f32>
```

Decomposes a 32-bit value into two 16-bit chunks, and reinterpets each chunk as a floating point value.
Component i of the result is the f32 representation of v, where v is the interpretation of bits 16×i through 16×i+15 of e as an [IEEE-754](https://www.w3.org/TR/WGSL/#biblio-ieee-754) binary16 value. See [§ 12.5.2 Floating Point Conversion](https://www.w3.org/TR/WGSL/#floating-point-conversion).

### PR DESCRIPTION
**Pull request by zoddicus**
Fixes #1293

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/expression/call/builtin/unpack2x16float.spec.ts [hunk 1]
```diff
@@ -6,15 +6,68 @@ interpretation of bits 16×i through 16×i+15 of e as an IEEE-754 binary16 value
...
+    const makeCase = (n: number): Case => {
+      n = quantizeToF32(n);
+      const results = unpack2x16float(n);
+      return {
+        input: [u32(n)],
+        expected: anyOf(...results.map(cmp)),
+      };
+    };
+
+    const cases: Array<Case> = fullF32Range().map(makeCase);
```
Comment (zoddicus) [inline src/webgpu/shader/execution/expression/call/builtin/unpack2x16float.spec.ts:70]: This should use fullU32Range(), not fullF32Range()

Comment (zoddicus) [reply src/webgpu/shader/execution/expression/call/builtin/unpack2x16float.spec.ts]: Done

**Comment by zoddicus** [comment]
This includes an implementation of fullU32Range, which ever unpack* PR lands first will add this function, and the others can be rebased to remove it from the change.

**Comment by zoddicus** [comment]
I have rewritten this to use the numerics framework. It also has been rebased onto https://github.com/gpuweb/cts/pull/1947, since it needs a fix from that PR to be correct.

The actual code to review is the changes in https://github.com/gpuweb/cts/pull/1934/commits/ff67443b845aa0b5481bebe2ccf6f393a2f8f13d

File: src/unittests/f32_interval.spec.ts [hunk 2]
```diff
@@ -2691,6 +2692,44 @@ interface PointToVectorCase {
...
+        { input: 0xbc00bc00, expected: [[-1], [-1]] },
+        { input: 0x49004900, expected: [[10], [10]] },
+        { input: 0xc900c900, expected: [[-10], [-10]] },
+
+        // f16 subnormals
+        { input: 0x000003ff, expected: [[0, kValue.f16.subnormal.positive.max], [0]] },
+        { input: 0x000083ff, expected: [[kValue.f16.subnormal.negative.min, 0], [0]] },
+
+        // f16 out of bounds
+        { input: 0x7c000000, expected: [kAny, kAny] },
```
Comment (ben-clayton) [inline src/unittests/f32_interval.spec.ts:2718]: Only one of the lanes here is OOB. Do we expect that to affect the other lane?

Comment (zoddicus) [reply src/unittests/f32_interval.spec.ts:2718]: Theoretically it shouldn't, but practically we have seen on the packing side drivers that mangle the non-OOB lane.

Comment (ben-clayton) [reply src/unittests/f32_interval.spec.ts:2718]: Gross. Okay then!

**Review by ben-clayton — APPROVED** [review]

File: src/unittests/f32_interval.spec.ts (hunk 2 above)
Comment (zoddicus) [reply src/unittests/f32_interval.spec.ts:2718]: There is probably a very depressing book to be written on all of the little gross things that happen in the graphics stack that make me sad :upside_down_face:

**Pull request merged**